Library/Homebrew/livecheck/livecheck.rb

```diff
@@ -394,42 +461,54 @@ def formula_name(formula, full_name: false)
+    end
+
     sig {
       params(
-        formula_or_cask: T.any(Formula, Cask::Cask),
-        status_str:      String,
-        messages:        T.nilable(T::Array[String]),
-        full_name:       T::Boolean,
-        verbose:         T::Boolean,
+        package_or_resource: T.any(Formula, Cask::Cask, Resource),
+        status_str:          String,
+        messages:            T.nilable(T::Array[String]),
+        full_name:           T::Boolean,
+        verbose:             T::Boolean,
       ).returns(Hash)
     }
-    def status_hash(formula_or_cask, status_str, messages = nil, full_name: false, verbose: false)
-      formula = formula_or_cask if formula_or_cask.is_a?(Formula)
-      cask = formula_or_cask if formula_or_cask.is_a?(Cask::Cask)
+    def status_hash(package_or_resource, status_str, messages = nil, full_name: false, verbose: false)
+      formula = package_or_resource if package_or_resource.is_a?(Formula)
+      cask = package_or_resource if package_or_resource.is_a?(Cask::Cask)
+      resource = package_or_resource if package_or_resource.is_a?(Resource)
 
       status_hash = {}
       if formula
         status_hash[:formula] = formula_name(formula, full_name: full_name)
       elsif cask
-        status_hash[:cask] = cask_name(formula_or_cask, full_name: full_name)
+        status_hash[:cask] = cask_name(package_or_resource, full_name: full_name)
+      elsif resource
+        status_hash[:resource] = resource_name(package_or_resource, full_name: full_name)
       end
       status_hash[:status] = status_str
       status_hash[:messages] = messages if messages.is_a?(Array)
 
       status_hash[:meta] = {
-        livecheckable: formula_or_cask.livecheckable?,
+        livecheckable: package_or_resource.livecheckable?,
       }
       status_hash[:meta][:head_only] = true if formula&.head_only?
 
       status_hash
     end
 
-    # Formats and prints the livecheck result for a formula.
-    sig { params(info: Hash, verbose: T::Boolean, ambiguous_cask: T::Boolean).void }
-    def print_latest_version(info, verbose:, ambiguous_cask: false)
-      formula_or_cask_s = "#{Tty.blue}#{info[:formula] || info[:cask]}#{Tty.reset}"
-      formula_or_cask_s += " (cask)" if ambiguous_cask
-      formula_or_cask_s += " (guessed)" if !info[:meta][:livecheckable] && verbose
+    # Formats and prints the livecheck result for a formula/cask/resource.
+    sig { params(info: Hash, verbose: T::Boolean, ambiguous_cask: T::Boolean, check_resource: T::Boolean).void }
+    def print_latest_version(info, verbose:, ambiguous_cask: false, check_resource: false)
+      package_or_resource_s = "  " if check_resource
+      package_or_resource_s += "#{Tty.blue}#{info[:formula] || info[:cask] || info[:resource]}#{Tty.reset}"
+      package_or_resource_s += " (cask)" if ambiguous_cask && !check_resource
+      package_or_resource_s += " (guessed)" if !info[:meta][:livecheckable] && !check_resource && verbose
```

Comment:
`info[:meta][:livecheckable]` is currently giving me an `undefined method `[]' for nil:NilClass` error when testing `influxdb` using the instructions in the PR description. This occurs when livecheck attempts to print information for a resource:

```
influxdb: 2.3.0 ==> 2.3.0
Error: influxdb: undefined method `[]' for nil:NilClass
```

To resolve this, you'll need to look into why `info[:meta]` is `nil` for a resource at this point. This may also explain why the `livecheckable` field is missing for resource objects in the verbose JSON output.

Reply:
Fixed the issue ✅